src/decoder/mod.rs

```diff
@@ -169,18 +172,30 @@ impl<R: Read> Decoder<R> {
         self.limits = limits;
     }
 
-    /// Reads all meta data until the first IDAT chunk
-    pub fn read_info(self) -> Result<Reader<R>, DecodingError> {
-        let mut streaming_decoder = StreamingDecoder::new();
-        streaming_decoder.set_decode_config(self.decode_config);
+    /// Read the PNG header and return the information contained within.
+    ///
+    /// Most image metadata will not be read until `read_info` is called, so those fields will be
+    /// None or empty.
+    pub fn read_header_info(&mut self) -> Result<&Info, DecodingError> {
+        while self.read_decoder.info().is_none() {
+            if self.read_decoder.decode_next(&mut Vec::new())?.is_none() {
+                return Err(DecodingError::Format(
+                    FormatErrorInner::MissingImageData.into(),
+                ));
+            }
```

Comment:
The specific error variant seems improper here. But maybe that's just our current usage if there is no better one.

Reply:
Happy to switch it to something else. I was just matching the current behavior: https://github.com/image-rs/image-png/blob/536f76f3660fde6ba8be876c7bac86bae7298cb9/src/decoder/mod.rs#L385-L389

Reply:
It seems more out-of-place now, especially if the method is being called explicitly. Maybe `UnexpectedEof`?